### PR TITLE
feat: support secret output for generate config

### PR DESCRIPTION
**Pull request by wtrocki**
This PR adds new features to generate configs

- Ability to put custom name 
- Extra env variables required for Service Registry API as workaround for:
https://github.com/Apicurio/apicurio-registry/issues/2466
- Support for secret type in generate config
```
apiVersion: v1
kind: Secret
metadata:
  name: default-1650902356
type: Opaque
stringData:
  ## Kafka Configuration
  KAFKA_HOST: [kannanrhoa-c-j-mf-ka---nol-u--g.bf2.kafka.rhcloud.com:443](http://kannanrhoa-c-j-mf-ka---nol-u--g.bf2.kafka.rhcloud.com:443/)
  ## Service Registry Configuration
  SERVICE_REGISTRY_URL: https://bu98.serviceregistry.rhcloud.com/t/5fec80da-9256-4334-a3b5-947994cf4b63
  SERVICE_REGISTRY_CORE_PATH: /apis/registry/v2
  SERVICE_REGISTRY_COMPAT_PATH: /apis/ccompat/v6
  
  ## Authentication Configuration
  RHOAS_CLIENT_ID: srvc-acct-f8f8d0c8-0e6e-486a-9fcd-2c284aba7436
  RHOAS_CLIENT_SECRET: 3d17fb47-115a-4bf5-ae93-d3483286d494
  RHOAS_OAUTH_TOKEN_URL: https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token
```

## Verification
```
minikube start
rhoas generate-config --type=secret --output-file=mysecret.yaml
oc create -f mysecret.yaml
```

### CODE REVIEW

**Comment by wtrocki** [comment]
I'm not 100% sure if layout of the generated files is okish.

**Comment by wtrocki** [comment]
CC @ppatierno . This fixes #1535

**Comment by wtrocki** [comment]
Not sure if SERVICE_REGISTRY_CORE_PATH and COmpat are good names. Looking for feedback. Feel free to change.

**Comment by ppatierno** [comment]
@wtrocki maybe I have an older version of rhoas but how this `generate-config` command is related to the service account creation command that returns json, env or properties file? How a context is related to a service account? I cannot find useful explanation in the doc.

**Comment by wtrocki** [comment]
Amazing feedback. 

Service account command is still there but it doesnt have required data like service urls. Context and generate-config are new commands. We are actually building guide for it and are in process of updating all docs

File: pkg/core/localize/locales/en/cmd/generate_config.en.toml [hunk 1]
```diff
@@ -13,6 +13,11 @@ $ rhoas generate-config --type json
 [generate.flag.type]
 one='Type of configuration file to be generated'
 
+[generate.common.flag.fileLocation.description]
+description = 'Description for --output-file flag'
+one = 'Sets a custom file location to save the credentials'
```
Comment (rkpattnaik780) [inline pkg/core/localize/locales/en/cmd/generate_config.en.toml:18]:
```suggestion
one = 'Sets a custom file location to save the configurations'
```

File: pkg/cmd/generate/templates.go [hunk 2]
```diff
@@ -2,34 +2,68 @@ package generate
...
-		{{if .KafkaHost}}KAFKA_HOST={{.KafkaHost}} 
-		{{end}}{{if .RegistryURL}}SERVICE_REGISTRY_URL={{.RegistryURL}}
-		{{end}}RHOAS_CLIENT_ID={{.ClientID}}
+		## Generated by rhoas cli
+		{{if .KafkaHost}}## Kafka Configuration
+		KAFKA_HOST={{.KafkaHost}}
+		{{end}}{{if .RegistryURL}}## Service Registry Configuration
+		SERVICE_REGISTRY_URL={{.RegistryURL}}
+		SERVICE_REGISTRY_CORE_PATH=` + registrycmdutil.REGISTRY_CORE_PATH + `
+		SERVICE_REGISTRY_COMPAT_PATH=` + registrycmdutil.REGISTRY_COMPAT_PATH + `
```
Comment (rkpattnaik780) [inline pkg/cmd/generate/templates.go:16]: Will it be better to provide the compatibility endpoint at once? As a user, I think I'll prefer to use one single env var over appending them myself in app.

Comment (rkpattnaik780) [reply pkg/cmd/generate/templates.go:16]: Ignore. Looks okay after seeing the [PR](https://github.com/redhat-developer/app-services-guides/pull/469) in guides repo.

File: pkg/cmd/generate/templates.go [hunk 3]
```diff
@@ -2,34 +2,68 @@ package generate
...
 `)
 
 	templateJSON = heredoc.Doc(`
 		{	
-			{{if .KafkaHost}}"kafkaHost":"{{.KafkaHost}}",
-			{{end}}{{if .RegistryURL}}"serviceRegistryUrl":"{{.RegistryURL}}",
-			{{end}}"rhoasClientID":"{{.ClientID}}",
-			"rhoasClientSecret":"{{.ClientSecret}}",
-			"rhoasOauthTokenUrl":"{{.TokenURL}}"
+            {{if .KafkaHost}}"kafkaHost":"{{.KafkaHost}}",
```
Comment (rkpattnaik780) [inline pkg/cmd/generate/templates.go:26]: Indentation for JSON and secrets file looks weird now.
```
{	
          "kafkaHost":"test-insta-c-j-fqn-ogaldf-coieg.bf2.kafka.rhcloud.com:443",
          "serviceRegistryUrl":"https:
```

**Review by rkpattnaik780 — COMMENTED** [review]
Verified. Some minor changes suggested.

File: pkg/cmd/generate/templates.go (hunk 2 above)
Comment (wtrocki) [reply pkg/cmd/generate/templates.go:16]: We need them separate as some tools require full uri and some just hostname

**Comment by ppatierno** [comment]
@wtrocki taking a look at the example you posted in the PR description, it seems to be clear to me that a "context" provides a way to define ... a context :-D ... around your Kafka instance which includes the URL to connect to and the service registry that you can get through the generate config commands. 
Still not clear why the `Secret` would contain just one set of credentials instead. If the user has more service account to be used against your Kafka instance (so in your context) shouldn't be all there inside the `Secret`? or shouldn't you specify somehow the service account to have credentials in the `Secret`?
By using the create service account command it's clear to me. You are getting credentials for that specific service account.

**Comment by wtrocki** [comment]
@rkpattnaik780 Thank you so much for fixes.

**Pull request merged**